modules/primo/includes/primo.search.inc

```diff
@@ -121,7 +125,7 @@ function primo_search_object_ids($ids) {
     }
   }
   else {
-    $translated_ids[$id] = variable_get('primo_sourceid') . $id;
+    $translated_ids[$ids] = variable_get('primo_sourceid') . $ids;
```

Comment:
:+1:

Another problem that could have been caught more easily with `array_map()` instead of `foreach` 😉 .